pkgs/by-name/ut/ut1999/package.nix

```diff
@@ -0,0 +1,122 @@
+{ lib, stdenv, requireFile, autoPatchelfHook, fetchurl,
+libgcc, wxGTK32, innoextract, makeWrapper,
+libGL, SDL2, openal, libmpg123, libxmp }:
+
+let
+  oldUnrealPatch = stdenv.mkDerivation {
+    name = "ut1999-oldunreal-patch";
+    version = "469d";
+    sourceRoot = ".";
+    src = fetchurl {
+      url = "https://github.com/OldUnreal/UnrealTournamentPatches/releases/download/v469d/OldUnreal-UTPatch469d-Linux-amd64.tar.bz2";
+      hash = "sha256-aoGzWuakwN/OL4+xUq8WEpd2c1rrNN/DkffI2vDVGjs=";
+    };
+
+    buildInputs = [
+      libgcc
+      stdenv.cc.cc
+      wxGTK32
+    ];
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+      makeWrapper
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      ls -alh
+      cp -r ./* $out
+
+      runHook postInstall
+    '';
+
+    dontBuild = true;
+  };
+
+  unpackGog = stdenv.mkDerivation {
+    name = "ut1999-gog";
+    version = "2.0.0.5";
+
+    src = requireFile rec {
+      name = "setup_ut_goty_2.0.0.5.exe";  # FIXME: use official name
+      sha256 = "00v8jbqhgb1fry7jvr0i3mb5jscc19niigzjc989qrcp9pamghjc";
+      message = ''
+        Unreal Tournament 1999 requires the official GOG package, version 2.0.0.5.
+
+        Once you download the file, run the following command:
+
+        nix-prefetch-url file://\$PWD/${name}
+      '';
+    };
+
+    buildInputs = [
+      innoextract
+    ];
+
+    unpackPhase = ''
+      innoextract --extract --exclude-temp "$src"
+    '';
+
+    installPhase = ''
+      mkdir $out
+      cp -r app/* $out
+    '';
+
+    dontBuild = true;
+    dontFixup = true;
+  };
+in stdenv.mkDerivation {
+  name = "ut1999";
+  version = "2.0.0.5";
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir $out
+
+    cp -r ${oldUnrealPatch}/* $out
+    chmod -R 755 $out
+
+    cp -r ${unpackGog}/Music $out
+    cp -r ${unpackGog}/Sounds $out
+    cp -n ${unpackGog}/Textures/* $out/Textures || true
+    cp -r ${unpackGog}/Maps $out
+
+    # Remove bundled libraries to use native versions instead
+    rm $out/System64/libfmod.so* \
+      $out/System64/libmpg123.so* \
+      $out/System64/libopenal.so* \
+      $out/System64/libSDL2* \
+      $out/System64/libxmp.so*
+
+    makeWrapper "$out/System64/ut-bin" "$out/bin/ut-bin" \
+      --suffix LD_LIBRARY_PATH : "${lib.strings.makeLibraryPath [
+        libGL
+        SDL2
+        openal
+        libmpg123
+        libxmp
+      ]}"
```

Comment:
Consider autoPatchelfHook instead.

Reply:
@Atemu , I've just realised I've put the `makeWrapper` in the wrong place (it should have been in the `oldUnrealPatch` sub-derivation). Nevertheless, `autoPatchElfHook` does not have an effect here. I think it's because the UT binary attempts to load them dynamically. The other dependencies (for example, `wxGTK32`) are caught by the elf hook.